packages/react/toast/src/Toast.tsx

```diff
@@ -512,7 +519,10 @@ const ToastImpl = React.forwardRef<ToastImplElement, ToastImplProps>(
                     });
                   }
                 })}
-              />
+              >
+                <VisuallyHidden>{renderLabel && context.label}</VisuallyHidden>
```

Comment:
We now trigger the announce in NVDA by updating this string in next tick, this means we can avoid rendering `children` in two places and solves our issue.